zerver/decorator.py

```diff
@@ -358,15 +363,16 @@ def _wrapped_func_arguments(
 
                     notify_bot_owner_about_invalid_json(user_profile, webhook_client_name)
                 elif isinstance(err, JsonableError) and not isinstance(
-                    err, UnsupportedWebhookEventType
+                    err, (UnsupportedWebhookEventType, AnomalousWebhookPayload)
                 ):
                     pass
                 else:
-                    if isinstance(err, UnsupportedWebhookEventType):
+                    if isinstance(err, (UnsupportedWebhookEventType, AnomalousWebhookPayload)):
                         err.webhook_name = webhook_client_name
                     log_exception_to_webhook_logger(
                         summary=str(err),
                         unsupported_event=isinstance(err, UnsupportedWebhookEventType),
+                        anomalous_payload=isinstance(err, AnomalousWebhookPayload),
```

Comment:
This sort of pattern really makes me wonder if `AnomalousWebhookPayload` should be a `UnsupportedWebhookEventType` subclass, or both derive from a common root class or protocol.  I don't have any namings that pop to mind for it immediately, though.

Reply:
That's a great idea, would certainly simplify the code that catches these webhook exceptions.